### PR TITLE
fix complex case for bregman

### PR DESCRIPTION
**Pull request by ziyiyin97**
as title

### CODE REVIEW

File: src/bregman.jl [hunk 1]
```diff
@@ -111,19 +111,20 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
...
             inds_z = findall(abs.(z) .> λ)
-            mul!(d, d, t)
-            d[:] .*= abs.(tk[:])/i
+            @views d[inds_z] .*= abs.(tk[:])/i
         end
         # Update z variable
         @. z = z + d
         # Get λ at first iteration
-        i == 1 && (λ = T(quantile(abs.(z), options.quantile)))
+        i == 1 && (λ = abs(T(quantile(abs.(z), options.quantile))))
```
Comment (mloubout) [inline src/bregman.jl:127]: don't see why would need `abs` here, it's `quantile(abs(z))` it's positive.

File: src/bregman.jl [hunk 2]
```diff
@@ -111,19 +111,20 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
...
 
         # Anti-chatter
         if options.antichatter
-            @. tk = tk - sign.(d)
+            @. tk = tk - sign(d)
             # Chatter correction
             inds_z = findall(abs.(z) .> λ)
-            mul!(d, d, t)
-            d[:] .*= abs.(tk[:])/i
+            @views d[inds_z] .*= abs.(tk[:])/i
```
Comment (mloubout) [inline src/bregman.jl]: wrong size that will error

File: src/bregman.jl [hunk 3]
```diff
@@ -111,19 +111,20 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
         d .= -TD*g
         # Step length
         t = (options.spg && i> 1) ? T(dot(x-xold, x-xold)/dot(x-xold, g-gold)) : T(options.alpha*f/norm(d)^2)
+        t = abs(t)
```
Comment (mloubout) [inline src/bregman.jl:114]: don't see how t can be not real or neagtive

**Comment by mloubout** [comment]
What case would that be?

File: src/bregman.jl (hunk 1 above)
Comment (ziyiyin97) [reply src/bregman.jl:127]: When T is complex then it's complex

File: src/bregman.jl (hunk 3 above)
Comment (ziyiyin97) [reply src/bregman.jl:114]: same as above

**Comment by mloubout** [comment]
You still didn't reply , what weird case leads to x being complex

**Pull request merged**